include/pgduckdb/pgduckdb_duckdb.hpp

```diff
@@ -2,6 +2,12 @@
 
 #include "duckdb.hpp"
 
+// FIXME: Make sure we don't need this
+extern "C" {
+#include "postgres.h"
+#include "access/xact.h"
+}
```

Comment:
Seems we need this for `XactEvent`. Sadly from what I can tell, this is an C enum so we can't forward declare it. It might work forward-declaring it as an `int` but it's a bit dangerous.

Since the callbacks are `static`, can we move them to a dedicated file/compilation unit? Would make splitting easier :-)